### DISCUSSION TITLE
would be very interesting to see the io500 benchmark of CORTX motr

### DISCUSSION THREAD

**Discussion by yanqingfu** [Seagate/cortx#435] [Q&A] [answered]
https://www.vi4io.org/io500/

https://github.com/VI4IO/io-500-dev

**Comment by johnbent** [comment, marked as answer]
Thanks @yanqingfu , I personally LOVE this idea.  I'm actually a cofounder of IO500!

Although CORTX has some very nice features for small IO and metadata such as our distributed key-value engine, note that it is designed for mass capacity storage and therefore might not be optimized for the full suite of behaviors exhibited in IO500.  However, it would be awesome to benchmark it and see just where it is.

Note also that IO500 has recently changed hosting and is now hosted at https://io500.org and the git repo is at https://github.com/io500/io500.

If you look at IOR (the main benchmark used in IO500), you'll see that it has a nicely modular design and has interfaces currently for RADOS and for DAOS so it would be cool to emulate one of those and create a direct to motr IOR interface.  Here for example is the DAOS one:
https://github.com/hpc/ior/blob/master/src/aiori-DFS.c

Here is the page where an ambitious person could learn about creating custom motr applications and could maybe use that info to create an IOR integration:
https://github.com/Seagate/cortx-motr/blob/main/doc/reading-list.md#motr-clients

If you want to try to do that, let me know and I'll do my best to assist!
